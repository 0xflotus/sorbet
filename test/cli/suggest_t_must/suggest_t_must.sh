#!/bin/bash
main/sorbet test/cli/suggest_t_must/suggest_t_must.rb 2>&1
