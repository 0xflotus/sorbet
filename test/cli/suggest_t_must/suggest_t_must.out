test/cli/suggest_t_must/suggest_t_must.rb:4: Method `[]` does not exist on `NilClass` component of `T.nilable(String)` http://go/e/7003
     4 |foo[0]
        ^^^^^^
  You could wrap it in `T.must()` before calling the function.
test/cli/suggest_t_must/suggest_t_must.rb:6: `T.nilable(String)` doesn't match `String` for argument `arg0` http://go/e/7002
     6 |"hi" + foo
        ^^^^^^^^^^
    https://git.corp.stripe.com/stripe-internal/sorbet/tree/master/rbi/core/string.rbi#L18: Method `+` has specified `arg0` as `String`
    18 |      arg0: String,
              ^^^^^
  Got T.nilable(String) originating from:
    test/cli/suggest_t_must/suggest_t_must.rb:3:
     3 |foo = T.let(nil, T.nilable(String))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  You could wrap it in `T.must()` before passing the argument.
Errors: 2
